homeassistant/components/sensor/envisalink.py

```diff
@@ -0,0 +1,68 @@
+"""
+Support for Envisalink sensors (shows panel info).
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.envisalink/
+"""
+import logging
+
+from homeassistant.components.envisalink import (EVL_CONTROLLER,
+                                                 EnvisalinkDevice,
+                                                 SIGNAL_KEYPAD_UPDATE)
+from homeassistant.util import convert
+
+REQUIREMENTS = ['pydispatcher==2.0.5']
+DEPENDENCIES = ['envisalink']
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """Perform the setup for Envisalink sensor devices."""
+    _configured_partitions = discovery_info['partitions']
+    add_devices_callback(
+        EnvisalinkSensor(convert(_configured_partitions[partNum]['name'],
+                                 str,
+                                 str.format("Partition #{0}", partNum)),
+                         partNum,
+                         EVL_CONTROLLER.alarm_state['partition'][partNum],
+                         EVL_CONTROLLER)
+        for partNum in _configured_partitions)
+
+
+class EnvisalinkSensor(EnvisalinkDevice):
+    """Representation of an envisalink keypad."""
+
+    def __init__(self, partitionName, partitionNumber, info, controller):
```

Comment:
I am surprised that none of the linters complained about the `camelCase` here. It should be `snake_case`